subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ExecuteActionsTaskExecuter.java

```diff
@@ -326,30 +315,22 @@ public void visitOutputProperties(OutputPropertyVisitor visitor) {
             for (OutputFilePropertySpec property : context.getTaskProperties().getOutputFileProperties()) {
                 File outputFile = property.getOutputFile();
                 if (outputFile != null) {
-                    visitor.visitOutputProperty(property.getPropertyName(), property.getOutputType(), outputFile);
+                    visitor.visitOutputProperty(property.getPropertyName(), property.getOutputType(), outputFile, property.getPropertyFiles());
                 }
             }
         }
 
         @Override
-        public void visitOutputTrees(CacheableTreeVisitor visitor) {
-            for (OutputFilePropertySpec property : context.getTaskProperties().getOutputFileProperties()) {
-                if (!(property instanceof CacheableOutputFilePropertySpec)) {
-                    throw new IllegalStateException("Non-cacheable property: " + property);
```

Comment:
Did this validation end up anywhere? Or was it already duplicated from `DefaultTaskCacheabilityResolver.shouldDisableCaching`?

Reply:
I added this in the past to make sure we never end up here. It was basically an assertion, but we don't need it. Indeed we should never get here because of `shouldDisableCaching`.